modules/youtube.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python
+'''
+youtube.py - Youtube Module
+
+Copyright 2014, Sujeet Akula (sujeet@freeboson.org)
+Copyright 2012, Dimitri Molenaars, Tyrope.nl.
+Copyright 2012-2013, Elad Alfassa, <elad@fedoraproject.org>
+Copyright 2012, Edward Powell, embolalia.net
+
+More info:
+ * jenni-misc: https://github.com/freeboson/jenni-misc/
+ * jenni: https://github.com/myano/jenni/
+ * Phenny: http://inamidst.com/phenny/
+'''
+
+
+
+import json
+import re
+from HTMLParser import HTMLParser
+import re, urllib, gzip, StringIO
+import web
+
+def title(bot, match):
+    """
+    Get information about the latest video uploaded by the channel provided.
+    """
+    if match is None:
+        return
+
+    uri = 'http://gdata.youtube.com/feeds/api/videos/' + match.group(2) + '?v=2&alt=json'
+
+    video_info = ytget(bot, None, uri)
+    if video_info is 'err':
+        return
+
+    #combine variables and print
+    message = '[YouTube] Title: ' + video_info['title'] + \
+              ' | Uploader: ' + video_info['uploader'] + \
+              ' | Uploaded: ' + video_info['uploaded'] + \
+              ' | Duration: ' + video_info['length'] + \
+              ' | Views: ' + video_info['views'] + \
+              ' | Comments: ' + video_info['comments'] + \
+              ' | Likes: ' + video_info['likes'] + \
+              ' | Dislikes: ' + video_info['dislikes'] + \
+              ' | Link: ' + video_info['link']
+
+    bot.say(HTMLParser().unescape(message))
+
+    return True
+
+def ytget(bot, trigger, uri):
+    try:
+        bytes = web.get(uri)
+        result = json.loads(bytes)
+        if 'feed' in result:
+            video_entry = result['feed']['entry'][0]
+        else:
+            video_entry = result['entry']
+    except:
+        bot.say('Something went wrong when accessing the YouTube API.')
```

Comment:
It would be cool to capture the exception and write it to the log, but the logging capabilities in jenni need to be enhanced significantly. Instead maybe capture the exception and spit out the exception type to the user.

```python
try:
  ...
except Exception as e:
  bot.say('Something went wrong when accessing the YouTube API: {0}'.format(e))
  return 'err'
```